### DISCUSSION TITLE
Boards manager installation option for 2.5.0 ?

### DISCUSSION THREAD

**Discussion by hmeijdam** [SpenceKonde/megaTinyCore#567] [Q&A]
I see you released 2.5.0 
Will there be a boards manager installation for it or will it be manual installation only?

Besides the Github release that I am using in a portable installation, I like to have another release version available, just in case something gets broken in the Github version while developing.

**Comment by SpenceKonde** [comment]
2.5.0? No, and 2.5.0 is being pulled 

I actually thought it was in board manager, but I'd forogotten to pull the new json file to the webserver. 

But 2.5.0 doesn't work, I just fixed the critical bug and am in the process of releasing 2.5.0 megaTinyCore and 1.4.0 DxCore

**Reply by SpenceKonde** [reply]
2.5.1.is now available in booard manager/

**Comment by hmeijdam** [comment]
Hmm.... for now I am getting this error if I want to update in Arduino IDE 1.8.16 portable installation via the boards manager.

![image](https://user-images.githubusercontent.com/9595465/146685493-402d64be-6f26-42bd-b2a2-586cb945b0f5.png)

Let me try an older IDE version.
[edit]  1.8.15 gives me the same error.

{edit 2]  FWIW, Installing your DXCore via boards manager in the same Arduino IDE 1.8.16 worked just fine

**Comment by SpenceKonde** [comment]
Am in the middle of moving atm might have a chance to look at it tonight is
probably something I screwed up while frantically trying to deal with the
bug before people showed up to help

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by hmeijdam** [reply]
No worries, this can wait. Having a house is more important than having a hobby.

**Comment by SpenceKonde** [comment]
It installed without issue for me

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Comment by hmeijdam** [comment]
I downloaded Arduino 1.8.18 into a fresh portable installation, and now this core installed without issues via the boards manager. So must be something on my end that's causing this. Sorry for bothering.

**Comment by fijam** [comment]
I could not update from 2.4.2 using Arduino IDE 1.8.15 or 1.8.18 either, with the same exception.

Had to delete the core manually from `%AppData%\Local\Arduino15\packages\megaTinyCore` and install it fresh.

**Reply by SpenceKonde** [reply]
Wtf is this? I haven't done anything that I'd expect to cause installation issues...

**Comment by hmeijdam** [comment]
Here is the error console message when attempting to update via boards manager
```

java.lang.NullPointerException
java.lang.RuntimeException: java.lang.NullPointerException
	at cc.arduino.contributions.packages.ui.ContributionManagerUI.lambda$onInstallPressed$2(ContributionManagerUI.java:179)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at cc.arduino.contributions.packages.ContributionInstaller.remove(ContributionInstaller.java:275)
	at cc.arduino.contributions.packages.ui.ContributionManagerUI.lambda$onInstallPressed$2(ContributionManagerUI.java:170)
	... 1 more

```

**Comment by hmeijdam** [comment]
The workaround to get it fixed is to delete the "megaTinyCore" folder and it's contents from the "Packages" folder. Then you can reinstall again from the Boards manager.

Note: I have tested on portable installation.

**Comment by SpenceKonde** [comment]
I don't know what is causing this issue and cannot reproduce it.

**Reply by hmeijdam** [reply]
I would not bother. I have about 12 portable installations of the IDE, and suddenly on 4 of them that have Megatinycore installed via boardmanager I could not upgrade Megatinycore anymore. The github version and the fresh (portable) install of 2.5.0 just work fine.
